src/StructuredLogViewer/Controls/BuildControl.xaml.cs

```diff
@@ -1024,11 +1030,30 @@ private object FindInFiles(string searchText, int maxResults, CancellationToken
                     return null;
                 }
 
-                var haystack = file.Value;
-                var resultsInFile = haystack.Find(searchText);
-                if (resultsInFile.Count > 0)
+                if (!string.IsNullOrEmpty(searchText) && searchText.StartsWith("$secret"))
                 {
-                    results.Add((file.Key, resultsInFile.Select(lineNumber => (lineNumber, haystack.GetLineText(lineNumber)))));
+                    var word = searchText.Replace("$secret", string.Empty).Trim();
+                    NodeQueryMatcher notMatcher = null;
+                    if (word.StartsWith("not(", StringComparison.OrdinalIgnoreCase) && word.EndsWith(")"))
+                    {
+                        word = word.Substring(4, word.Length - 5);

```

Comment:
Super nit pick
```suggestion
                        word = word.Substring("not(".Length, word.Length - "not()".Length);

```
or a comment explaining the magic numbers